### PR TITLE
media service

### PR DESCRIPTION
**Pull request by NguyenTienVinh**

### CODE REVIEW

File: soundcloud_2/app/src/main/java/com/framgia/soundcloud_2/localsong/LocalSongsFragment.java [hunk 1]
```diff
@@ -104,6 +116,7 @@ public void start() {
         mRecyclerView.setLayoutManager(new LinearLayoutManager(getActivity()));
         mRecyclerView.setItemAnimator(new DefaultItemAnimator());
         mRecyclerView.setAdapter(mSongOfflineAdapter);
+        mDatabaseManager = new DatabaseManager(getActivity());
```
Comment (vtabk2) [inline soundcloud_2/app/src/main/java/com/framgia/soundcloud_2/localsong/LocalSongsFragment.java]: Singleton  
http://www.androiddesignpatterns.com/2012/05/correctly-managing-your-sqlite-database.html

File: soundcloud_2/app/src/main/java/com/framgia/soundcloud_2/service/PlayerService.java [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+                break;
+            case EVENT_NEXT:
+                mTransportControls.skipToNext();
+                break;
+            case EVENT_PREVIOUS:
+                mTransportControls.skipToPrevious();
+                break;
+            case EVENT_STOP:
+                mTransportControls.stop();
+                break;
```
Comment (vtabk2) [inline soundcloud_2/app/src/main/java/com/framgia/soundcloud_2/service/PlayerService.java]: thiếu default

File: soundcloud_2/app/src/main/java/com/framgia/soundcloud_2/service/PlayerService.java [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+    public void loadSong() {
+        mListTrack = mDatabaseHelper.getListTrack();
+        mAudioIndex = loadAudioIndex(getApplicationContext());
+        if (mAudioIndex == -1 || mAudioIndex > mListTrack.size())
+            mAudioIndex = 0;
+        mTrack = mListTrack.get(mAudioIndex);
+    }
+
+    public void initMediaPlayer() {
+        if (mMediaPlayer == null)
```
Comment (vtabk2) [inline soundcloud_2/app/src/main/java/com/framgia/soundcloud_2/service/PlayerService.java]: convention 2.4
https://github.com/framgia/coding-standards/blob/master/eng/android/codingstyleguide.md

File: soundcloud_2/app/src/main/java/com/framgia/soundcloud_2/utils/Constant.java [hunk 4]
```diff
@@ -17,7 +17,14 @@
     }
 
     public static class KeyIntent {
+        public static final String EVENT_PLAY_NEW_AUDIO = " EVENT_PLAY_NEW_AUDIO";
         public static final String EXTRA_CATEGORY = "category";
+        public static final String BROADCAST_UPDATE_CONTROL = "UpdateControl";
+        public static final String EVENT_PLAY = "EVENT_PLAY";
```
Comment (DoanVanToan) [inline soundcloud_2/app/src/main/java/com/framgia/soundcloud_2/utils/Constant.java]: e xem lại cách đặt tên Action 
https://github.com/framgia/coding-standards/blob/master/eng/android/codingstyleguide.md
static final String ACTION_OPEN_USER = "com.myapp.action.ACTION_OPEN_USER";

File: soundcloud_2/app/src/main/java/com/framgia/soundcloud_2/localsong/LocalSongsFragment.java [hunk 5]
```diff
@@ -77,6 +82,12 @@ public void onDataNotAvailable() {
     @Override
     public void playMusic(int songIndex) {
         // TODO: Play music
+        DatabaseManager.getInstance(getActivity()).clearListTrack();
```
Comment (DoanVanToan) [inline soundcloud_2/app/src/main/java/com/framgia/soundcloud_2/localsong/LocalSongsFragment.java]: Khai báo biến DatabaseManager instance variable

File: soundcloud_2/app/src/main/java/com/framgia/soundcloud_2/utils/Constant.java [hunk 6]
```diff
@@ -18,6 +19,14 @@
 
     public static class KeyIntent {
         public static final String EXTRA_CATEGORY = "category";
+        public static final String BROADCAST_UPDATE_CONTROL = "UpdateControl";
+        public static final String EVENT_PLAY_NEW_AUDIO = "com.soundcloud_2.action" +
```
Comment (DoanVanToan) [inline soundcloud_2/app/src/main/java/com/framgia/soundcloud_2/utils/Constant.java]: Nên đặt tên là action. 
BROADCAST_UPDATE_CONTROL > ACTION_UPDATE_CONTROL

File: soundcloud_2/app/src/main/java/com/framgia/soundcloud_2/utils/Constant.java [hunk 7]
```diff
@@ -18,6 +19,14 @@
 
     public static class KeyIntent {
         public static final String EXTRA_CATEGORY = "category";
+        public static final String ACTION_UPDATE_CONTROL = "UpdateControl";
```
Comment (DoanVanToan) [inline soundcloud_2/app/src/main/java/com/framgia/soundcloud_2/utils/Constant.java]: Đặt giá trị cho Action chưa đúng

**Pull request merged**